### PR TITLE
exp(x)

### LINKED ISSUE #83: number.eo:303-305: min. Implement exp using this taylor...
The puzzle `56-a1660657` from #56 has to be resolved: 

https://github.com/objectionary/eo-math/blob/a61f115a10a787c2ff52a411e483aaf678e3a7e5/src/main/eo/org/eolang/math/number.eo#L303-L305

The puzzle was created by levBagryansky on 10-Nov-22. 

Estimate: 90 minutes,  role: DEV. 

If you have any technical questions, don't ask me, submit new tickets instead. The task will be \"done\" when the problem is fixed and the text of the puzzle is _removed_ from the source code. Here is more about [PDD](http://www.yegor256.com/2009/03/04/pdd.html) and [about me](http://www.yegor256.com/2017/04/05/pdd-in-action.html).

### PR DESCRIPTION
**Pull request by levBagryansky**
closes: #83 
`number.exp` implemented

### CODE REVIEW

**Comment by levBagryansky** [comment]
@includealex please review

File: src/main/eo/org/eolang/math/number.eo [hunk 1]
```diff
@@ -30,6 +30,8 @@
 +version 0.0.0
 
 [n] > number
+  n > @
```
Comment (includealex) [inline src/main/eo/org/eolang/math/number.eo:33]: @levBagryansky do we really need this addition?

File: src/main/eo/org/eolang/math/number.eo [hunk 2]
```diff
@@ -300,9 +310,46 @@
...
+          exp-x0
+          exp-x1
+    as-float > value!
+    0.69314718055994530941 > ln-2!
+    as-int. > m!
+      number
+        div.
+          value
+          number.exp.ln-2
+    15000000.0 > count!
```
Comment (includealex) [inline src/main/eo/org/eolang/math/number.eo:337]: @levBagryansky what is the origin of this magical number? Why don't you use 1e6 or 1e9?

File: src/main/eo/org/eolang/math/number.eo [hunk 3]
```diff
@@ -68,15 +70,23 @@
       ^.n.as-bytes
       7F-C0-00-00
 
+  # Checking if number is positive-infinity
+  [] > is-positive-infinity
+    eq. > @
```
Comment (includealex) [inline src/main/eo/org/eolang/math/number.eo:75]: @levBagryansky is it a good idea to create new object `is-positive-infinity`?

File: src/main/eo/org/eolang/math/number.eo [hunk 4]
```diff
@@ -68,15 +70,23 @@
...
+      ^.n > value!
+      positive-infinity
+
+  # Checking if number is negative-infinity
+  [] > is-negative-infinity
+    eq. > @
```
Comment (includealex) [inline src/main/eo/org/eolang/math/number.eo:81]: The same

**Review by includealex — COMMENTED** [review]
@levBagryansky good one, just a few comments below.

File: src/main/eo/org/eolang/math/number.eo (hunk 1 above)
Comment (levBagryansky) [reply src/main/eo/org/eolang/math/number.eo:33]: @includealex Yes, this allows to use `n`'s attributes in object `(number n)`. For instance, without it you can not call `(number 1).neg`. Check how `text` works:
https://github.com/objectionary/eo-strings/blob/master/src/main/eo/org/eolang/txt/text.eo

File: src/main/eo/org/eolang/math/number.eo (hunk 3 above)
Comment (levBagryansky) [reply src/main/eo/org/eolang/math/number.eo:75]: @includealex Yes it is used in `number.exp` and in `number-tests`. If we need `is-infinite` object we need theese objects too.

File: src/main/eo/org/eolang/math/number.eo (hunk 2 above)
Comment (levBagryansky) [reply src/main/eo/org/eolang/math/number.eo:337]: @includealex This number provides the good accuracy.

**Review by levBagryansky — COMMENTED** [review]
@includealex answered

**Review by includealex — APPROVED** [review]
@levBagryansky Thanks!

**Comment by levBagryansky** [comment]
@includealex please check, I removed redundant `is-positive-infinity`

**Review by includealex — APPROVED** [review]
@levBagryansky I like it. @yegor256 please take a look

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**